test/ductile/conn_test.clj

```diff
@@ -15,63 +15,84 @@
-  (is (= {:as :json
-          :content-type :json
-          :throw-exceptions false
-          :connection-manager cm
-          :query-params {:param-1 "param-1"}}
-         (sut/make-http-opts cm opts [:param-1])))
-  (is (= {:as :json
-          :content-type :json
-          :throw-exceptions false
-          :connection-manager cm}
-         (sut/make-http-opts cm opts [])
-         (sut/make-http-opts cm opts nil)))
-  (is (= {:as :json
-          :content-type :json
-          :throw-exceptions false
-          :query-params {:param-1 "param-1"}}
-         (sut/make-http-opts nil opts [:param-1])))))
+              :param-2 "param-2"}
+        common {:as :json
+                :content-type :json
+                :throw-exceptions false}]
+
+    (is (= (assoc common
+                  :connection-manager cm
+                  :query-params (dissoc opts :param-2))
+           (sut/make-http-opts conn-wo-auth
+                               opts
+                               [:refresh
+                                :_source
+                                :param-1])))
+    (is (= (assoc common
+                  :connection-manager cm
+                  :query-params (dissoc opts :param-2)
+                  :form-params doc)
+            (sut/make-http-opts conn-wo-auth
+                                opts
+                                [:refresh
+                                 :_source
+                                 :param-1]
+                                doc
+                                nil)))
+    (is (= (assoc common
+                  :connection-manager cm
+                  :query-params (dissoc opts :param-2)
+                  :body doc)
+           (sut/make-http-opts conn-wo-auth
+                               opts
+                               [:refresh
+                                :_source
+                                :param-1]
+                               nil
+                               doc)))
+    (is (= (assoc common
+                  :connection-manager cm
+                  :query-params {:param-1 "param-1"})
+           (sut/make-http-opts conn-wo-auth opts [:param-1])))
+    (is (= (assoc common
+                  :connection-manager cm)
+           (sut/make-http-opts conn-wo-auth opts [])
+           (sut/make-http-opts conn-wo-auth opts nil)))
+    (is (= (assoc common
+                  :query-params {:param-1 "param-1"})
+           (sut/make-http-opts nil opts [:param-1])))))
+
+(deftest testing-auth-params
+  (testing "auth options should be properly prepared and integrated in http params."
+    (let [test-auth (fn [auth-params expected]
+                      (let [{:keys [auth] :as conn}
+                            (sut/connect {:host "127.0.0.1"
+                                          :port 9200
+                                          :auth auth-params})
+                            auth-http-opts (-> (sut/make-http-opts conn {})
```

Comment:
You could also exercise that wrong creds result in a connection not being granted

Reply:
I [added an integration test](https://github.com/threatgrid/ductile/pull/9/commits/946196660b846db555fbc0518060261e8511cc3f) with authorized and unauthorized requests with missing auth, basic auth and API key.